Moda.Web/src/moda.web.reactclient/src/app/components/common/timeline/moda-timeline.tsx

```diff
@@ -18,27 +18,28 @@ import {
   ModaDataGroup,
   ModaDataItem,
   ModaTimelineOptions,
-  RangeItemTemplateProps,
 } from '.'
 import { ModaEmpty } from '..'
 import { getLuminance } from '@/src/utils/color-helper'
+import { ItemTemplate } from '@/src/app/components/common/timeline/types'
+import { ThemeName } from '@/src/app/components/contexts/theme/types'
 
 const { Text } = Typography
 
-export type ModaTimelineProps<TI = any, TG = any> = {
-  data: ModaDataItem<TI>[]
-  groups?: ModaDataGroup<TG>[]
+export type ModaTimelineProps<T> = {
+  data: ModaDataItem<T>[]
+  groups?: ModaDataGroup<T>[]
   isLoading: boolean
-  options: ModaTimelineOptions
-  rangeItemTemplate?: (props: RangeItemTemplateProps<TI>) => JSX.Element
-  groupTemplate?: (props: GroupTemplateProps<TG>) => JSX.Element
+  options: ModaTimelineOptions<T>
+  rangeItemTemplate?: ItemTemplate<T>
+  groupTemplate?: ItemTemplate<T>
   emptyMessage?: string
 }
 
-const RangeItemTemplate = (props: RangeItemTemplateProps<ModaDataItem>) => {
-  // TODO: the 0.6 needs to be tested with some of the other colors
+const RangeItemTemplate: ItemTemplate<unknown> = (props) => {
+  // TODO: The 0.6 needs to be tested with some of the other colors
   const fontColor =
-    getLuminance(props.item.itemColor) > 0.6 ? '#4d4d4d' : '#FFFFFF'
+    getLuminance((props.item as ModaDataItem).itemColor ?? props.foregroundColor ?? '') > 0.6 ? '#4d4d4d' : '#FFFFFF'
```

Comment:
`ModaDataItem.itemColor` was/is optional. This default template seems to depend on this optional property. Unsure if we want to make this property non optional, or if we want a fallback here.